### PR TITLE
feat(expr): add `array_max`

### PR DESCRIPTION
**Pull request by xzhseh**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

add `array_max`
related issue #11996

## Checklist

- [ ] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by xzhseh** [comment]
The logic is pretty much the same with `array_min`, wondering if we could encapsulate the two (`array_min/max`) together.

File: src/expr/src/vector_op/array_max.rs
Comment (wangrunji0408) [inline src/expr/src/vector_op/array_max.rs:1]: I prefer to put this function in the same module as `array_min` because they are almost the same.

**Review by wangrunji0408 — COMMENTED** [review]
LGTM! let's wait for #12071 to be merged.

File: src/expr/src/vector_op/array_max.rs
Comment (xzhseh) [reply src/expr/src/vector_op/array_max.rs:1]: Thanks for the advice! I'll put them together later.

File: src/expr/src/vector_op/array_min_max.rs [hunk 1]
```diff
@@ -36,3 +38,22 @@ pub fn array_min<T: Scalar>(list: ListRef<'_>) -> Result<Option<T>> {
...
+#[function("array_max(list) -> int256")]
+#[function("array_max(list) -> date")]
+#[function("array_max(list) -> time")]
+#[function("array_max(list) -> timestamp")]
+#[function("array_max(list) -> timestamptz")]
+#[function("array_max(list) -> varchar")]
+#[function("array_max(list) -> bytea")]
+pub fn array_max<T: Scalar>(list: ListRef<'_>) -> Result<Option<T>> {
+    let min_value = list.iter().flatten().map(DefaultOrdered).max();
+    match min_value.map(|v| v.0).to_owned_datum() {
```
Comment (wangrunji0408) [inline src/expr/src/vector_op/array_min_max.rs:55]:
```suggestion
    let max_value = list.iter().flatten().map(DefaultOrdered).max();
    match max_value.map(|v| v.0).to_owned_datum() {
```

**Review by wangrunji0408 — APPROVED** [review]
LGTM! Thanks

**Pull request merged**